src/server/journal/journal_test.cc

```diff
@@ -125,5 +126,74 @@ TEST(Journal, WriteRead) {
   }
 }
 
+TEST(Journal, PendingBuf) {
+  PendingBuf pbuf;
+
+  ASSERT_TRUE(pbuf.empty());
+  ASSERT_EQ(pbuf.size(), 0);
+
+  pbuf.push("one");
+  pbuf.push(" small");
+  pbuf.push(" test");
+
+  ASSERT_FALSE(pbuf.empty());
+  ASSERT_EQ(pbuf.size(), 14);
+
+  {
+    auto& sending_buf = pbuf.PrepareSendingBuf();
+    ASSERT_EQ(sending_buf.buf.size(), 3);
+    ASSERT_EQ(sending_buf.mem_size, 14);
+
+    ASSERT_EQ(sending_buf.buf[0], "one");
+    ASSERT_EQ(sending_buf.buf[1], " small");
+    ASSERT_EQ(sending_buf.buf[2], " test");
+  }
+
+  const size_t string_num = 2000;
+  for (size_t i = 0; i < string_num; ++i) {
+    pbuf.push("big_test");
```

Comment:
Let's have some of the strings larger than SSO
Also above - please replace one of "one", " small", " test" with a larger than SSO string